src/klein/_headers_compat.py

```diff
@@ -0,0 +1,86 @@
+# -*- test-case-name: klein.test.test_headers -*-
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+Support for interoperability with L{twisted.web.http_headers.Headers}.
+"""
+
+from typing import AnyStr, Iterable, Text, Tuple
+
+from attr import attrib, attrs
+from attr.validators import instance_of
+
+from twisted.web.http_headers import Headers
+
+from zope.interface import implementer
+
+from ._headers import (
+    IMutableHTTPHeaders, RawHeaders, String, headerNameAsBytes,
+    headerValueAsText, normalizeHeaderName, rawHeaderName,
+    rawHeaderNameAndValue,
+)
+
+AnyStr, Iterable, RawHeaders, String, Tuple  # Silence linter
+
+
+__all__ = ()
+
+
+
+@implementer(IMutableHTTPHeaders)
+@attrs(frozen=True)
+class HTTPHeadersWrappingHeaders(object):
+    """
+    HTTP entity headers.
+
+    This is an L{IMutableHTTPHeaders} implementation that wraps a L{Headers}
+    object.
+    """
+
+    _headers = attrib(validator=instance_of(Headers))  # type: Headers
+
+
+    @property
+    def rawHeaders(self):
+        # type: () -> RawHeaders
+        def pairs():
+            # type: () -> Iterable[Tuple[bytes, bytes]]
+            for name, values in self._headers.getAllRawHeaders():
+                name = normalizeHeaderName(name)
+                for value in values:
+                    yield (name, value)
+
+        return tuple(pairs())
+
+
+    def getValues(self, name):
+        # type: (AnyStr) -> Iterable[AnyStr]
+        if isinstance(name, bytes):
+            values = self._headers.getRawHeaders(name, default=())
+        elif isinstance(name, Text):
+            values = (
+                headerValueAsText(value)
+                for value in self._headers.getRawHeaders(
+                    headerNameAsBytes(name), default=()
+                )
+            )
+        else:
+            raise TypeError("name {!r} must be text or bytes".format(name))
+
+        return tuple(values)
```

Comment:
I'm not a big fan of using tuple as "immutable list". Especially typing wise. Since the value is fresh anyway, why not return a list?

Reply:
Right, it's spec'ed as an `Iterable`, so a `list` should be fine.

Reply:
d772564c7425ec705126b1db386ad4abec77f788